sigmastate/src/main/scala/org/ergoplatform/ErgoAddress.scala

```diff
@@ -226,12 +230,15 @@ case class ErgoAddressEncoder(networkPrefix: NetworkPrefix) {
           val p = GroupElementSerializer.parse(r)
           new P2PKAddress(ProveDlog(p), contentBytes)
         case Pay2SHAddress.addressTypePrefix =>
+          if (contentBytes.length != 24) { //192-bits hash used
+            throw new Exception(s"Improper content in P2SH script: $addrStr")
+          }
           new Pay2SHAddress(contentBytes)
```

Comment:
@kushti, @greenhat  I'm not sure we don't have potential problem with HF here. Are we sure this method is not used during contract verification? Because otherwise we are changing  the semantics of scripts (some of them may start failing).

Reply:
@aslesarenko addresses are not known on the protocol (ErgoTree) level, so I do not see any problems. Also, 192-bits input was expected in the P2SH script, now the corresponding check added to the address instantiation logic